### PR TITLE
feat(core): Bump to Souin v1.6.2

### LINKED ISSUE #19: Cache handler module on https://caddyserver.com/download
Are there any plans to add this Caddy module to the download page at https://caddyserver.com/download?
Thanks!

### LINKED ISSUE #21: Possible to configure a disk based cache instead of in-memory?
I'm not sure if this is something that badger or olric can do, but I'm looking to cache pages to disk using the cache handler and then serve them from there, rather than keeping it in memory for the ttl.

Is this something that this plugin can do? If so, are there any docs or examples for this? I followed the link to the official badger docs, but I didn't see anything for this specifically.

Thank you!

### LINKED ISSUE #22: Bug with rewrite directive?
I found a behaviour that feels weird to me: The module looks for cached responses with the original request URI, but if there is a `rewrite` internally the cache entry is written with the rewritten URI. That means that original request URI does not get a cache hit if requested again.

Minimal example:

```caddyfile
{
	order cache before rewrite
}

http://:80 {

	cache {
		ttl 20s
	}
	rewrite /say/* /respond{path}
	respond /respond/* "{path.2}"

}
```

If I request `http://localhost/say/hi` several times I would think that I get the cached response from the 2nd request onward (within the 20 seconds of course). What actually happens, is that I always get a cache miss.
Requesting `http://localhost/respond/say/hi` as the second request yields me a cache hit.

I know I could just `order cache after rewrite` for this silly use case but I have a more elaborate use case in combination with my [image filter module](https://github.com/ueffel/caddy-imagefilter).

<details>
<summary>elaborate use case</summary>

```caddyfile
{
	order cache before rewrite
	order image_filter before file_server
}

http://:80 {
	@thumbnail {
		path_regexp thumb (?i)/w([0-9]+)(/.+)$
	}
	root .
	handle @thumbnail {
		cache {
			ttl 20s
		}
		rewrite {re.thumb.2}
		image_filter {
			resize {re.thumb.1} 0
		}
	}
}
```

It resizes an image where size is given within the path. So `http://localhost/w100/image.png` would give me a 100 pixel wide thumbnail of `image.png`. Because of the cache misses when `order`ing `cache before rewrite` the image is resized on every request.
If I `order cache after rewrite` here and first request `http://localhost/w100/image.png` and then `http://localhost/w200/image.png` I actually get the 100 pixel wide cached response in the 2nd request even though I should get a newly resized 200px wide response image.

</details>

### LINKED ISSUE #25: Caddy returns 200 when 5xx fom upstream
When I use 'cache' handler, Caddy always returns status code 200 and empty response whenever upstream is unavailable or returns 5xx errors.
Please advice how can upstream errors can be passed through the cache. Thanks.

### PR DESCRIPTION
**Pull request by darkweak**
Closes #19 
Closes #21 
Closes #22 
Closes #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,23 +69,87 @@ respond "Hello World!"
...
+        }
+    }
 }
 
 cache @souin-api {}
 ```
 What does these directives mean?  
 |  Key                      |  Description                                                                                                                                 |  Value example                                                                                                          |
 |:--------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------|
+| `allowed_http_verbs`      | The HTTP verbs allowed to be cached                                                                                                          | `GET POST PATCH`<br/><br/>`(default: GET HEAD)`                                                                         |
```
Comment (dunglas) [inline README.md:152]: Can you elaborate on this? IMHO we should stick with the spec.

**Comment by francislavoie** [comment]
I think we should have the people who reported the linked issues to try this before closing the issues. Are we sure their problems are fixed or are you just guessing?

Re #21, my suggestion hasn't been addressed, i.e. using go-humanize to parse size values in the Caddyfile. I think that's very important to do, to aid with ease of use and for consistency with config elsewhere in the Caddyfile.

File: README.md (hunk 1 above)
Comment (darkweak) [reply README.md:152]: It's to support the GraphQL partial cache introduced in the previous Souin version. IMHO we should not put aside the GraphQL cache support in the cache-handler.

Comment (darkweak) [reply README.md:152]: It was the first thing we thought before introduce this but after discussions with some users, they have the use case to restrict the GraphQL cache to POST and PATCH only and we cannot tell to the end user "if you want to enable the GraphQL cache, you have to send only a POST/PATCH request".

**Comment by darkweak** [comment]
As I said in the https://github.com/darkweak/souin/pull/190#issuecomment-1065830215 this is a low-level configuration. I prefer to be as close as possible to the Badger declaration but a PR is under preparation to use go-humanize to parse sizes directly in their Options.

**Comment by francislavoie** [comment]
If you're satisfied with the changes, IMO you can go ahead and merge it. Probably best to just get people to try the changes and report if they have any issues with it.

But I don't have push/merge rights on this repo, so I don't really have any authority here 😛

**Comment by darkweak** [comment]
@francislavoie I already asked them to try on the Souin repository before applying changes here and they confirmed
that works and as cache-handler and the caddy middleware integration in Souin are equal it's good 😄

**Pull request merged**